build/platform-android.mk

```diff
@@ -5,7 +5,7 @@ SHAREDLIBSUFFIX = so
 SHAREDLIBSUFFIXFULLVER=$(SHAREDLIBSUFFIX)
 SHAREDLIBSUFFIXMAJORVER=$(SHAREDLIBSUFFIX)
 SHLDFLAGS =
-NDKLEVEL = 12
+NDKLEVEL = $(shell echo $(TARGET) | grep -E -o "[0-9]+")
```

Comment:
yes, the user normally only pass e.g. TARGET=android-21 but does not pass the NDKLEVEL. 
1. default NDKLEVEL=12 maybe too old.
2. and I think pass TARGET=andorid-21 but actually use NDKLEVEL=12 is bit strage.

Reply:
Well there might be one reason for it; currently TARGET is used as the target for the java code. For the java side of android apps, you generally should set the target SDK version to the highest available, while you set the minimum SDK version to the minimum you ever want it to run on. For native code, you can only set one version, the minimum. So therefore it might make sense to keep them separate since you generally should not build with the NDKLEVEL set to the highest.

I guess this tries to fix the fact that newer NDK versions don't contain all the old platform versions any longer? In that case I think it might be better to just bump up the default value to whichever is the minimum supported by current NDK versions.

Reply:
ok, let the NDKLEVEL is not changed. will fix this.